app/src/main/java/sensors_in_paradise/sonar/page2/LoggingManager.kt

```diff
@@ -148,6 +148,16 @@ class LoggingManager(
     }
 
     private fun startLogging() {
+        while (labels.size > 1) {
+            labels.removeAt(0)
+        }
```

Comment:
Labels are cleared in `stopLogging()`, why do we need this?

Reply:
Hm, okay, I see (when labels are picked during no active recording, the labels array is still filled). I think a comment would be helpful if not extract the new code into a new method so that it is

```
recordingStartTime = System.currentTimeMillis()
initializeLabels(recordingStartTime)
```

Wdyt?

all in all this method is pretty complex and tightly coupled

Reply:
👍